packages/firestore/test/unit/remote/datastore.test.ts

```diff
@@ -0,0 +1,197 @@
+    const datastore = newDatastore(
+      new EmptyCredentialsProvider(),
+      new MockConnection(),
+      serializer
+    );
+    datastore.terminate();
+    await expect(invokeDatastoreImplInvokeRpc(datastore))
+      .to.eventually.be.rejectedWith(/terminated/i)
+      .and.have.property('code', Code.FAILED_PRECONDITION);
+  });
+
+  it('DatastoreImpl.invokeRPC() rethrows a FirestoreError', async () => {
+    const connection = new MockConnection();
+    connection.invokeRPC = () =>
+      Promise.reject(new FirestoreError(Code.ABORTED, 'zzyzx'));
+    const credentials = new MockCredentialsProvider();
+    const datastore = newDatastore(credentials, connection, serializer);
+    await expect(invokeDatastoreImplInvokeRpc(datastore))
+      .to.eventually.be.rejectedWith('zzyzx')
+      .and.have.property('code', Code.ABORTED);
+    expect(credentials.invalidateTokenInvoked).to.be.false;
+  });
+
+  it('DatastoreImpl.invokeRPC() wraps unknown exceptions in a FirestoreError', async () => {
+    const connection = new MockConnection();
+    connection.invokeRPC = () => Promise.reject('zzyzx');
+    const credentials = new MockCredentialsProvider();
+    const datastore = newDatastore(credentials, connection, serializer);
+    await expect(invokeDatastoreImplInvokeRpc(datastore))
+      .to.eventually.be.rejectedWith('zzyzx')
+      .and.have.property('code', Code.UNKNOWN);
+    expect(credentials.invalidateTokenInvoked).to.be.false;
+  });
+
+  it('DatastoreImpl.invokeRPC() invalidates the token if unauthenticated', async () => {
+    const connection = new MockConnection();
+    connection.invokeRPC = () =>
+      Promise.reject(new FirestoreError(Code.UNAUTHENTICATED, 'zzyzx'));
+    const credentials = new MockCredentialsProvider();
+    const datastore = newDatastore(credentials, connection, serializer);
+    await expect(invokeDatastoreImplInvokeRpc(datastore))
+      .to.eventually.be.rejectedWith('zzyzx')
+      .and.have.property('code', Code.UNAUTHENTICATED);
+    expect(credentials.invalidateTokenInvoked).to.be.true;
+  });
+
+  it('DatastoreImpl.invokeStreamingRPC() fails if terminated', async () => {
+    const datastore = newDatastore(
+      new EmptyCredentialsProvider(),
+      new MockConnection(),
+      serializer
+    );
+    datastore.terminate();
+    await expect(invokeDatastoreImplInvokeStreamingRPC(datastore))
+      .to.eventually.be.rejectedWith(/terminated/i)
+      .and.have.property('code', Code.FAILED_PRECONDITION);
+  });
+
+  it('DatastoreImpl.invokeStreamingRPC() rethrows a FirestoreError', async () => {
+    const connection = new MockConnection();
+    connection.invokeStreamingRPC = () =>
+      Promise.reject(new FirestoreError(Code.ABORTED, 'zzyzx'));
+    const credentials = new MockCredentialsProvider();
+    const datastore = newDatastore(credentials, connection, serializer);
+    await expect(invokeDatastoreImplInvokeStreamingRPC(datastore))
```

Comment:
This doesn't seem to verify that we return a FirestoreError. I think you can do this as such:
 
```
await expect(invokeDatastoreImplInvokeStreamingRPC(datastore)).to.be.eventually.rejectedWIth(new FirestoreError(Code.ABORTED, 'zzyzx'))
``` 

If that doesn't work, you can use a try/catch here.

This also applies to other tests such as"DatastoreImpl.invokeRPC() wraps unknown exceptions in a FirestoreError". Since GRPC errors also have codes, I suspect that these tests may have passed even without your change.